src/lib/components/workflow/filter-search/datetime-filter.svelte

```diff
@@ -239,18 +152,117 @@
               bind:hour={startHour}
               bind:minute={startMinute}
               bind:second={startSecond}
-              bind:half={startHalf}
+              twelveHourClock={false}
             />
-            <div class="flex justify-end">
-              <Button variant="ghost" on:click={onApply}
-                >{translate('apply')}</Button
+          </div>
+        </MenuItem>
+        <MenuDivider />
+        <MenuItem>
+          <div class="flex flex-col gap-2">
+            <DatePicker
+              label={translate('end')}
+              on:datechange={onEndDateChange}
+              selected={endDate}
+              todayLabel={translate('today')}
+              closeLabel={translate('close')}
+              clearLabel={translate('clear-input-button-label')}
+            />
+            <TimePicker
+              bind:hour={endHour}
+              bind:minute={endMinute}
+              bind:second={endSecond}
+              twelveHourClock={false}
+            />
+          </div>
+        </MenuItem>
+      {:else}
+        <MenuItem on:click={() => (type = 'relative')}>
+          <div class="flex flex-col">
+            <label
+              class="flex flex-row items-center gap-2 cursor-pointer"
+              for="relative-time"
+            >
+              <input
+                on:click|stopPropagation={() => {
+                  type = 'relative';
+                }}
+                class="w-4 h-4 accent-gray-900"
+                type="radio"
+                checked={type === 'relative'}
+                id="relative-time"
+                tabindex="-1"
+              />
+              {translate('relative')}
+            </label>
```

Comment:
I did look into this a bit, but I think it might best if we addressed refactoring these in a separate PR. We have things like the `<div/>` (aka `input-container`), etc. that complicate the styling here.